v2/internal/genericarmclient/generic_client.go

```diff
@@ -84,6 +86,30 @@ func NewGenericClientFromHTTPClient(cloudCfg cloud.Configuration, creds azcore.T
 	// the value IN the interface IS nil).
 	if httpClient != nil {
 		opts.Transport = httpClient
+	} else {
+		// If httpClient is not provided, we use a HTTPClient with default Transport + settings
+		// to establish multiple TCP ARMClient connections to avoid throttling.
+		// TODO: Use https://github.com/Azure/go-armbalancer here once its prod ready.
+		httpTransport := &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
```

Comment:
Ahh one more thing -- we should have a single global one of these, not 1 per call to this function (that's how the SDK does it as well, see the `transport_default_http_client` link I gave below.

Reply:
So rather than having a `httpClient` per `GenericClient`, we want to share a default `httpClient` across all `GenericClients`?

Reply:
Share the transport, the HTTPClient can be constructed 1 for each.